apps/emqx_gateway/src/emqx_gateway_conf.erl

```diff
@@ -208,7 +210,7 @@ listeners(GwName0) ->
     ),
     Listeners = emqx_utils_maps:jsonable_map(
         emqx_utils_maps:deep_get(
-            [<<"gateway">>, GwName, <<"listeners">>], RawConf
+            [<<"gateway">>, GwName, <<"listeners">>], RawConf, []
```

Comment:
Since the type of this config key is a map, maybe it'd be more consistent to return a map as a default value (then handling an empty list in `convert_listeners` won't be needed either):

```suggestion
            [<<"gateway">>, GwName, <<"listeners">>], RawConf, #{}
```